http://sourceware.org/ml/gdb-cvs/2012-05/msg00010.html

### src/gdb/ChangeLog	2012/05/02 17:13:39	1.14194
### src/gdb/ChangeLog	2012/05/02 20:00:31	1.14195
## -1,3 +1,8 @@
+2012-05-02  Jan Kratochvil  <jan.kratochvil@redhat.com>
+
+	Fix --without-auto-load-safe-path for MS-Windows host platform.
+	* auto-load.c (filename_is_in_dir): Return 1 for DIR_LEN 0.
+
 2012-05-02  Eli Zaretskii  <eliz@gnu.org>
 
 	* gdb_curses.h: Undefine KEY_EVENT before including curses
--- src/gdb/auto-load.c	2012/04/17 15:56:21	1.4
+++ src/gdb/auto-load.c	2012/05/02 20:00:36	1.5
@@ -231,6 +231,12 @@
   while (dir_len && IS_DIR_SEPARATOR (dir[dir_len - 1]))
     dir_len--;
 
+  /* Ensure auto_load_safe_path "/" matches any FILENAME.  On MS-Windows
+     platform FILENAME even after gdb_realpath does not have to start with
+     IS_DIR_SEPARATOR character, such as the 'C:\x.exe' filename.  */
+  if (dir_len == 0)
+    return 1;
+
   return (filename_ncmp (dir, filename, dir_len) == 0
 	  && (IS_DIR_SEPARATOR (filename[dir_len])
 	      || filename[dir_len] == '\0'));
